frontend/src/views/producer/dashboard/edition-form/Editor.jsx

```diff
@@ -49,82 +60,137 @@ const RedButton = styled(Button)`
   }
 `;
 
-const defaulTicketTypeValue = { ...TicketCatalog.create().toJSON() };
-
-const Editor = observer(
-  class Editor extends React.Component {
-    eventDraft = observable({
-      ...EventModel.create({ ticketCatalog: [defaulTicketTypeValue] }).toJSON(),
-    });
-
-    componentDidMount() {
-      const { palette } = this.props.theme;
-      // override specific class name from ant-design
-      // THIS IS A HACK
-      injectGlobal`
+class Editor extends React.Component {
```

Comment:
This component is quite big. Could you consider splitting it to smaller, more manageable components?

Reply:
YES! This component is the definition of WIP. I whole heartedly aggrees I should pull it out.
But budget is tight and there is still consumer to be done, so I will add a todo here, and will revisit to refactor it later on. Sounds good for you?

Reply:
sounds good enough 👍